staging/src/k8s.io/apiserver/pkg/endpoints/filters/traces.go

```diff
@@ -22,19 +22,17 @@ import (
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 	"go.opentelemetry.io/otel/trace"
 
-	"k8s.io/component-base/traces"
+	tracing "k8s.io/component-base/tracing"
 )
 
 // WithTracing adds tracing to requests if the incoming request is sampled
-func WithTracing(handler http.Handler, tp *trace.TracerProvider) http.Handler {
+func WithTracing(handler http.Handler, tp trace.TracerProvider) http.Handler {
 	opts := []otelhttp.Option{
-		otelhttp.WithPropagators(traces.Propagators()),
+		otelhttp.WithPropagators(tracing.Propagators()),
 		otelhttp.WithPublicEndpoint(),
+		otelhttp.WithTracerProvider(tp),
 	}
-	if tp != nil {
```

Comment:
in TracingOptions#ApplyTo, when TracingOptions is nil or no config file is specified, we can still be left with a nil TracerProvider instead of a no-op one, right? in that case, wouldn't we still need the nil check here and the conditional WithTracerProvider append?

Reply:
I _think_ this: 
https://github.com/kubernetes/kubernetes/blob/ba7e26a2d7ad34a8d5f2d19b168a571fae869b31/staging/src/k8s.io/apiserver/pkg/server/config.go#L375
ensures TracerProvider will always be the Noop even when config is nil
so will never be nil here:
https://github.com/kubernetes/kubernetes/blob/ba7e26a2d7ad34a8d5f2d19b168a571fae869b31/staging/src/k8s.io/apiserver/pkg/server/config.go#L856 

@dashpole I'd appreciate another look at it, though, please

Reply:
ok, I missed the initialization there

Reply:
I'll take another pass.  As a side note, once we update to v1 of the otel libraries (blocked by etcd upgrade), passing a `nil` tracer provider will be valid, and will default to the no-op provider.